app/src/main/java/io/homeassistant/companion/android/controls/HaControl.kt

```diff
@@ -49,6 +54,22 @@ interface HaControl {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             control.setAuthRequired(authRequired)
         }
+        if (entity.attributes["icon"]?.toString()?.startsWith("mdi:") == true &&
+            !entity.attributes["icon"]?.toString()?.substringAfter(":").isNullOrBlank()
+        ) {
+            val iconName = entity.attributes["icon"]!!.toString().split(':')[1]
+            val iconDrawable =
+                IconicsDrawable(context, "cmd-$iconName").apply {
+                    sizeDp = 48
+                }
+            val colorTint = when (entity.state) {
+                "on" -> if (entity.domain == "light") "#FDD663" else "#8AB4F8"
+                "off", "unavailable", "unknown" -> "#A6A6A7"
+                else -> "#8AB4F8"
+            }
```

Comment:
Why not use color resources to give it a more descriptive name and a preview in Android Studio next to the line numbers?

Reply:
great suggestion, thank you 🙏